File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> ans = new HashSet<>();
        int i = 0 , j = 0;
        int maxsize = 0 , size = 0;
        while(i < s.length() && j < s.length()){
            if(!ans.contains(s.charAt(j))){
                size++;
                ans.add(s.charAt(j));
                maxsize = Math.max(size , maxsize);
                j++;
            }
            else{
                size--;
                ans.remove(s.charAt(i));
                i++;
            }
        }
        return maxsize;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = set()
        i = 0 
        j = 0
        size = 0 
        maxsize = 0
        while j < len(s):
            if s[j] not in ans:
                size += 1
                ans.add(s[j])
                j += 1
                maxsize = max(j - i, maxsize)
            else:
                size -= 1
                ans.remove(s[i])
                i += 1
        return maxsize        
                